### PR TITLE
Disable version check restriction

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Allow to disable the version migration check restriction, useful for testing migration, especially by QA.

## Related issues

slack https://camunda.slack.com/archives/CSQ2E3BT4/p1734357369901219

See also migration testing for Camunda Exporter and Identity, etc.

### CODE REVIEW

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java [hunk 1]
```diff
@@ -66,7 +81,7 @@ public long getMaxAppendBatchSizeInBytes() {
    * @deprecated Deprecated in favor of {@link RaftCfg#getFlush()}. The equivalent is a null
    *     configuration, e.g. {@link new FlushConfig(null)}. Will be removed in 8.3.0.
    */
-  @Deprecated(since = "8.2.0", forRemoval = true)
+  @Deprecated(since = "8.2.0", forRemoval = DEFAULT_VERSION_CHECK_ENABLED)
```
Comment (npepinpe) [inline zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java:84]: :x: This looks like automated refactoring? Doesn't make much sense imo :smile:

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java [hunk 2]
```diff
@@ -76,7 +91,7 @@ public boolean isDisableExplicitRaftFlush() {
    *     call {@link RaftCfg#setFlush(FlushConfig)} with a null configuration, e.g. {@code new
    *     FlushConfig(null)}. Will be removed in 8.3.0.
    */
-  @Deprecated(since = "8.2.0", forRemoval = true)
+  @Deprecated(since = "8.2.0", forRemoval = DEFAULT_VERSION_CHECK_ENABLED)
```
Comment (npepinpe) [inline zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java:94]: :x: same here

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java [hunk 3]
```diff
@@ -32,9 +39,17 @@ public class ExperimentalCfg implements ConfigurationEntry {
...
-
   private FeatureFlagsCfg features = new FeatureFlagsCfg();
 
+  public boolean isVersionCheckRestrictionEnabled() {
+    return versionCheckRestrictionEnabled;
+  }
+
+  public void setVersionCheckRestrictionEnabled(final boolean versionCheckRestrictionEnabled) {
+    this.versionCheckRestrictionEnabled = versionCheckRestrictionEnabled;
+    return;
```
Comment (npepinpe) [inline zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java:50]: :question: Why the return?

**Review by npepinpe — APPROVED** [review]
:x: Please add it to the `.template` files as well :pray: 

No need for another review, just address all comments

**Review by lenaschoenburg — APPROVED** [review]

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisKujawa) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/configuration/ExperimentalCfg.java:94]: Interesting how did this happened

**Pull request merged**